lib/compress/index.js

```diff
@@ -1336,6 +1336,7 @@ function tighten_body(statements, compressor) {
                 || node instanceof AST_Destructuring
                 || node instanceof AST_Expansion
                    && node.expression instanceof AST_Symbol
+                   && node.expression.definition()
```

Comment:
Very interesting. I had always assumed every AST_Symbol had a definition.

I had a look at the inheritance tree, and it looks like `AST_This` is also a symbol (technically). So maybe we can change the instanceof check in the line above to check for `AST_SymbolRef` which is more specifically referring to a reference.

If that also fixes your code, then probably somewhere in your unminified bundle there's a `...this` which is breaking this code!